### PR TITLE
starting to flesh out the service access layer and tools

### PR DESCRIPTION
**Pull request by natechols**
With a functioning stub for 'pbservice'.

### CODE REVIEW

File: smrt-server-tools/src/main/scala/com/pacbio/secondary/smrttools/client/ServiceAccessLayer.scala [hunk 1]
```diff
@@ -43,10 +43,17 @@ class ServiceAccessLayer(val baseUrl: URL)(implicit actorSystem: ActorSystem) {
...
   def serviceStatusPipeline: HttpRequest => Future[ServiceStatus] = sendReceive ~> unmarshal[ServiceStatus]
+  def getDataSetByUuidPipeline: HttpRequest => Future[DataSetMetaDataSet] = sendReceive ~> unmarshal[DataSetMetaDataSet]
 
   val statusUrl = toUrl("/status")
 
   def getStatus: Future[ServiceStatus] = serviceStatusPipeline {
     Get(statusUrl)
   }
+
+  // FIXME this should take either an Int or a UUID, but how?
```
Comment (natechols) [inline smrt-server-tools/src/main/scala/com/pacbio/secondary/smrttools/client/ServiceAccessLayer.scala]: So how is this done?  I couldn't figure out how to make Either[Int, UUID] work

File: smrt-server-tools/src/main/scala/com/pacbio/secondary/smrttools/tools/PbService.scala [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    } children(
+      arg[Int]("dataset-id") required() action { (i, c) =>
+        c.copy(dataset_id = i)
+      } text "Dataset ID"
+    ) text "Show dataset details"
+  }
+}
+
+
+object PbServiceApp extends App {
```
Comment (natechols) [inline smrt-server-tools/src/main/scala/com/pacbio/secondary/smrttools/tools/PbService.scala]: This is just awful, I'm not sure what the right way to do this is (of course most of the actual functionality is missing too)

File: smrt-server-tools/src/main/scala/com/pacbio/secondary/smrttools/client/ServiceAccessLayer.scala (hunk 1 above)
Comment (mpkocher) [inline smrt-server-tools/src/main/scala/com/pacbio/secondary/smrttools/client/ServiceAccessLayer.scala]: The standard java/scala style is to use camelcase. We should try to follow that model.

Comment (mpkocher) [inline smrt-server-tools/src/main/scala/com/pacbio/secondary/smrttools/client/ServiceAccessLayer.scala]: There's a way to do this. However, for now can you use explicit getDataSetByUUID(uuid: UUID)

String sub can use s"${a}/$b" format.

**Comment by natechols** [comment]
Tripped over .gitignore changes, will try again.

**Pull request closed without merging**